libs/libvtrutil/test/test_geometry.cpp

```diff
@@ -20,108 +20,143 @@ TEST_CASE("Point", "[vtr_geometry/Point]") {
+            REQUIRE(r1.ymin() == pi_1.y());
+            REQUIRE(r1.ymax() == pi_2.y());
+        }
 
-    SECTION("equality") {
-        REQUIRE(r1 == r2);
-    }
+        SECTION("point_accessors") {
+            REQUIRE(r1.bottom_left() == pi_1);
+            REQUIRE(r1.top_right() == pi_2);
+            REQUIRE(r2.bottom_left() == pi_1);
+            REQUIRE(r2.top_right() == pi_2);
+        }
 
-    SECTION("location") {
-        REQUIRE(r1.xmin() == pi_1.x());
-        REQUIRE(r1.xmax() == pi_2.x());
-        REQUIRE(r1.ymin() == pi_1.y());
-        REQUIRE(r1.ymax() == pi_2.y());
-    }
+        SECTION("dimensions") {
+            REQUIRE(r1.width() == 5);
+            REQUIRE(r1.height() == 8);
+            REQUIRE(r2.width() == 5);
+            REQUIRE(r2.height() == 8);
+        }
 
-    SECTION("point_accessors") {
-        REQUIRE(r1.bottom_left() == pi_1);
-        REQUIRE(r1.top_right() == pi_2);
-        REQUIRE(r2.bottom_left() == pi_1);
-        REQUIRE(r2.top_right() == pi_2);
-    }
+        SECTION("contains_int") {
+            REQUIRE(r2.contains(pi_1));
+            REQUIRE(r2.contains({6, 4}));
+            REQUIRE_FALSE(r2.contains({100, 4}));
+            REQUIRE_FALSE(r2.contains(pi_2));
+            REQUIRE(vtr::Rect<int>(pi_1).contains(pi_1));
+        }
 
-    SECTION("dimensions") {
-        REQUIRE(r1.width() == 5);
-        REQUIRE(r1.height() == 8);
-        REQUIRE(r2.width() == 5);
-        REQUIRE(r2.height() == 8);
-    }
+        SECTION("strictly_contains_int") {
+            REQUIRE_FALSE(r2.strictly_contains(pi_1));
+            REQUIRE(r2.strictly_contains({6, 4}));
+            REQUIRE_FALSE(r2.strictly_contains({100, 4}));
+            REQUIRE_FALSE(r2.strictly_contains(pi_2));
+        }
 
-    SECTION("contains_int") {
-        REQUIRE(r2.contains(pi_1));
-        REQUIRE(r2.contains({6, 4}));
-        REQUIRE_FALSE(r2.contains({100, 4}));
-        REQUIRE_FALSE(r2.contains(pi_2));
-    }
+        SECTION("coincident_int") {
+            REQUIRE(r2.coincident(pi_1));
+            REQUIRE(r2.coincident({6, 4}));
+            REQUIRE_FALSE(r2.coincident({100, 4}));
+            REQUIRE(r2.coincident(pi_2));
+        }
 
-    SECTION("strictly_contains_int") {
-        REQUIRE_FALSE(r2.strictly_contains(pi_1));
-        REQUIRE(r2.strictly_contains({6, 4}));
-        REQUIRE_FALSE(r2.strictly_contains({100, 4}));
-        REQUIRE_FALSE(r2.strictly_contains(pi_2));
-    }
+        SECTION("bounds_int") {
+            REQUIRE(r1 == bounding_box(r3, r4));
+        }
+
+        SECTION("empty_int") {
+            REQUIRE(vtr::Rect<int>().empty());
+        }
 
-    SECTION("coincident_int") {
-        REQUIRE(r2.coincident(pi_1));
-        REQUIRE(r2.coincident({6, 4}));
-        REQUIRE_FALSE(r2.coincident({100, 4}));
-        REQUIRE(r2.coincident(pi_2));
+        SECTION("sample_int") {
+            REQUIRE(sample(vtr::Rect<int>(pi_1, pi_2), 0, 0, 17) == pi_1);
```

Comment:
Probably worth having one test that isn't at the boundary?

Reply:
Done.